view/frontend/web/instantsearch.js

```diff
@@ -207,408 +206,454 @@ requirejs(['algoliaBundle','Magento_Catalog/js/price-utils'], function(algoliaBu
 						}
 					}
 				}
 			],
 			/**
-			 * Search box
-			 * Docs: https://community.algolia.com/instantsearch.js/v2/widgets/searchBox.html
+			 * searchBox
+			 * Docs: https://www.algolia.com/doc/api-reference/widgets/search-box/js/
 			 **/
 			searchBox: {
 				container: instant_selector,
-				placeholder: algoliaConfig.translations.searchFor
+				placeholder: algoliaConfig.translations.searchFor,
+				showSubmit: false
 			},
 			/**
-			 * Stats
-			 * Docs: https://community.algolia.com/instantsearch.js/v2/widgets/stats.html
+			 * stats
+			 * Docs: https://www.algolia.com/doc/api-reference/widgets/stats/js/
 			 **/
 			stats: {
 				container: '#algolia-stats',
 				templates: {
-					body: $('#instant-stats-template').html()
-				},
-				transformData: function (data) {
-					data.first = data.page * data.hitsPerPage + 1;
-					data.last = Math.min(data.page * data.hitsPerPage + data.hitsPerPage, data.nbHits);
-					data.seconds = data.processingTimeMS / 1000;
-					
-					data.translations = window.algoliaConfig.translations;
-					
-					return data;
+					text: function (data) {
+						var hoganTemplate = algoliaBundle.Hogan.compile($('#instant-stats-template').html());
+						var _data = Object.assign({}, data)
+
+						_data.first = data.page * data.hitsPerPage + 1;
+						_data.last = Math.min(data.page * data.hitsPerPage + data.hitsPerPage, data.nbHits);
+						_data.seconds = data.processingTimeMS / 1000;
+						_data.translations = window.algoliaConfig.translations;
+
+						return hoganTemplate.render(_data)
+					}
 				}
 			},
 			/**
-			 * Sorting
-			 * Docs: https://community.algolia.com/instantsearch.js/v2/widgets/sortBySelector.html
+			 * sortBy
+			 * Docs: https://www.algolia.com/doc/api-reference/widgets/sort-by/js/
 			 **/
-			sortBySelector: {
+			sortBy: {
 				container: '#algolia-sorts',
-				indices: algoliaConfig.sortingIndices,
-				cssClass: 'form-control'
+				items: algoliaConfig.sortingIndices.map(function (sortingIndice) {
+					return {
+						label: sortingIndice.label,
+						value: sortingIndice.name,
+					}
+				})
 			},
 			/**
-			 * Widget name: Current refinements
+			 * currentRefinements
 			 * Widget displays all filters and refinements applied on query. It also let your customer to clear them one by one
-			 * Docs: https://community.algolia.com/instantsearch.js/v2/widgets/currentRefinedValues.html
+			 * Docs: https://www.algolia.com/doc/api-reference/widgets/current-refinements/js/
 			 **/
-			currentRefinedValues: {
+			currentRefinements: {
 				container: '#current-refinements',
-				cssClasses: {
-					root: 'facet'
-				},
 				templates: {
-					header: '<div class="name">' + algoliaConfig.translations.selectedFilters + '</div>',
-					clearAll: algoliaConfig.translations.clearAll,
 					item: $('#current-refinements-template').html()
 				},
-				attributes: attributes,
-				onlyListedAttributes: true
+				panelOptions: {
```

Comment:
is this because you wrap everything in a panel now? I'm missing some context here probably :)

Reply:
Yes, if `panelOptions` is present:

https://github.com/algolia/algoliasearch-magento-2/blob/98d9457f8d46f33819d562a0ae4fbf4ba3069771/view/frontend/web/instantsearch.js#L640-L642

Reply:
Yes, everything facet widget is wrapped in a panel.
To reuse the current abstraction I has to "bundle" the panel configs along with the widget config.
This are some of the main things I thinking about removing with the refactoring, but for now this migrate the template.header